### PR TITLE
Setting sample rate: got rid of memory leak

### PR DESCRIPTION
**Pull request by srietkerk**
In trying to set the output sample rate, I added a new channel to the data pipeline every time that `checkEnv()` was called, thus causing programs to explode in size. With the new CODAL apis, I can set the sample rate of the existing channel and thus avoid a memory leak.

Closes https://github.com/microsoft/pxt-microbit/issues/5131

Note: This change in functionality is now activating the mic on the hardware. This is something we might want to change.

### CODE REVIEW

**Review by thsparks — APPROVED** [review]
LGTM! I had one question, but not a huge deal either way. Might still be good to get a review from someone familiar with the APIs though :)

**Review by jwunderl — APPROVED** [review]
Good to dedup as thomas mentioned but otherwise LGTM

File: libs/audio-recording/recording.cpp [hunk 1]
```diff
@@ -156,15 +151,23 @@ void setInputSampleRate(int sampleRate) {
...
+    }
 }
 
 /**
  * Set the sample rate for both input and output
 */
 //%
 void setBothSamples(int sampleRate) {
-    checkEnv(sampleRate);
+    if (recording == NULL) {
```
[the remark this answers is not in the document]
Comment (srietkerk) [reply libs/audio-recording/recording.cpp:166]: Yeah, it definitely can. I was  uncertain with the goals to reduce code size, if calling a function creates larger programs than just having the if statements where they're needed, but I will make this change. Thanks, Thomas!

[the remark this answers is not in the document]
Comment (jwunderl) [reply libs/audio-recording/recording.cpp:166]: Even without the code size coming into play, reducing two code paths that share code & are conceptually doing the same thing to one is beneficial in general~

**Pull request merged**